linera-service/src/chain_listener.rs

```diff
@@ -152,6 +152,7 @@ where
             client
         };
         let (listener, _listen_handle, mut local_stream) = client.listen().await?;
+        client.synchronize_from_validators().await?;
```

Comment:
If this empirically made a difference, I suspect the race condition is still there.

Reply:
It should, yes: We first start listening, and then we synchronize. So it shouldn't be possible for a block to be added after synchronizing but before listening.
(The other way round is fine: If we add a block during synchronization and then get a notification about it.)